src/app/core/admin/admin-entity-details/admin-entity-field/default-value-options/default-value-options.component.spec.ts

```diff
@@ -0,0 +1,182 @@
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(DefaultValueOptionsComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+
+    component.entityType = Entity;
+    component.inheritedFieldSelectElement = jasmine.createSpyObj(["open"]);
+  });
+
+  it("should create", () => {
+    expect(component).toBeTruthy();
+  });
+
+  it("should reset form fields when mode field is changed", () => {
+    const newMode = "dynamic";
+    component.form.get("mode").setValue(newMode);
+
+    expect(component.form.get("value").value).toBeNull();
+    expect(component.form.get("localAttribute").value).toBeNull();
+    expect(component.form.get("field").value).toBeNull();
+    expect(component.mode).toBe(newMode);
+  });
+
+  it("should auto-select static mode when value is added and mode is undefined", () => {
+    component.mode = undefined; // Simulate mode not being set
+    component.form.get("value").setValue("Some value");
+    expect(component.mode).toEqual("static");
+  });
+
+  it("should emit valueChange event when changed form is valid (static mode)", () => {
+    spyOn(component.valueChange, "emit");
+    component.form.setValue({
+      mode: "static",
+      value: "New value",
+      localAttribute: null,
+      field: null,
+    });
+    expect(component.valueChange.emit).toHaveBeenCalledWith(
+      jasmine.objectContaining({ mode: "static", value: "New value" }),
+    );
+  });
+  it("should emit valueChange event when changed form is valid (dynamic mode)", () => {
+    spyOn(component.valueChange, "emit");
+    component.form.setValue({
+      mode: "dynamic",
+      value: "New value",
+      localAttribute: null,
+      field: null,
+    });
+    expect(component.valueChange.emit).toHaveBeenCalledWith(
+      jasmine.objectContaining({ mode: "dynamic", value: "New value" }),
+    );
+  });
+  it("should emit valueChange event when changed form is valid (inherited mode)", () => {
+    spyOn(component.valueChange, "emit");
+    component.form.setValue({
+      mode: "inherited",
+      value: null,
+      localAttribute: "localAttribute",
+      field: "field",
+    });
+    expect(component.valueChange.emit).toHaveBeenCalledWith(
+      jasmine.objectContaining({
+        mode: "inherited",
+        localAttribute: "localAttribute",
+        field: "field",
+      }),
+    );
+  });
+  it("should not emit valueChange event when changed form is invalid", () => {
+    spyOn(component.valueChange, "emit");
+    component.form.setValue({
+      mode: "static",
+      value: null,
+      localAttribute: null,
+      field: null,
+    });
+    expect(component.valueChange.emit).not.toHaveBeenCalled();
+
+    component.form.setValue({
+      mode: "inherited",
+      value: null,
+      localAttribute: "foo",
+      field: null,
+    });
+    // TODO: why is this a valid form with the required "field" value missing?
+    expect(component.valueChange.emit).not.toHaveBeenCalled();
```

Comment:
for some reason the form (both in tests and in actual app) first is valid and in a second run invalid, so one change is emitted ...